src/yb/util/ulimit_info.cc

```diff
@@ -0,0 +1,87 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+// The following only applies to changes made to this file as part of YugaByte development.
+//
+// Portions Copyright (c) YugaByte, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations
+// under the License.
+//
+
+#include "yb/util/ulimit_info.h"
+
+#include <string>
+#include <sstream>
+#include <sys/resource.h>
+
+#include "yb/util/env.h"
+
+using std::string;
+using std::stringstream;
+
+namespace yb {
+
+static stringstream& getLimit(
+    stringstream& ss, const char* pfx, const char* sfx, int resource, int rightshift) {
+  ss << "ulimit: " << pfx << " ";
+  int64_t cur, max;
+  Status status = Env::Default()->GetUlimit(resource, &cur, &max);
+  if (status.ok()) {
+    if (cur == RLIM_INFINITY)
```

Comment:
Minor style comment: the overall if has braces while there are internals ifs with single lines below. This makes it slightly difficult to track the if corresponding to the else. I would suggest adding braces to all the if and else in this block to make it easier to track by matching the braces.